src/views/Checkout.vue

```diff
@@ -202,59 +202,59 @@ export default class Checkout extends Vue {
-
-        const request: KeyguardClient.SignTransactionRequest = {
-            layout: 'checkout',
-            shopOrigin: this.rpcState.origin,
-            appName: this.request.appName,
-            shopLogoUrl: this.request.shopLogoUrl,
-
-            keyId: walletInfo.keyId,
-            keyPath: accountInfo!.path,
-            keyLabel: walletInfo.label,
-
-            sender: (contractInfo || accountInfo!).address.serialize(),
-            senderType: contractInfo ? contractInfo.type : Nimiq.Account.Type.BASIC,
-            senderLabel: (contractInfo || accountInfo!).label,
-            recipient: this.request.recipient.serialize(),
-            recipientType: this.request.recipientType,
-            // recipientLabel: '', // Checkout is using the shopOrigin instead
-            value: this.request.value,
-            fee: this.request.fee,
-            validityStartHeight,
-            data: this.request.data,
-            flags: this.request.flags,
-        };
-
-        staticStore.keyguardRequest = request;
-
-        const client = this.$rpc.createKeyguardClient();
-        client.signTransaction(request);
+        // proceed to transaction signing
+        switch (senderAccount.type) {
+            case WalletType.LEGACY:
+            case WalletType.BIP39:
+                if (!this.height) return;
+
+                // The next block is the earliest for which tx are accepted by standard miners
+                const validityStartHeight = this.height + 1
+                    - TX_VALIDITY_WINDOW
+                    + this.request.validityDuration;
+
+                const request: KeyguardClient.SignTransactionRequest = {
+                    layout: 'checkout',
+                    shopOrigin: this.rpcState.origin,
+                    appName: this.request.appName,
+                    shopLogoUrl: this.request.shopLogoUrl,
+
+                    keyId: senderAccount.keyId,
+                    keyPath: signer!.path,
+                    keyLabel: senderAccount.label,
+
+                    sender: (senderContract || signer).address.serialize(),
+                    senderType: senderContract ? senderContract.type : Nimiq.Account.Type.BASIC,
+                    senderLabel: (senderContract || signer).label,
+                    recipient: this.request.recipient.serialize(),
+                    recipientType: this.request.recipientType,
+                    // recipientLabel: '', // Checkout is using the shopOrigin instead
+                    value: this.request.value,
+                    fee: this.request.fee,
+                    validityStartHeight,
+                    data: this.request.data,
+                    flags: this.request.flags,
+                };
+
+                staticStore.keyguardRequest = request;
+                const client = this.$rpc.createKeyguardClient();
+                client.signTransaction(request);
+                return;
+            case WalletType.LEDGER:
```

Comment:
You could move the LEDGER case to the top, as it's very short and would be better for visibility, I think.